### DISCUSSION TITLE
Which form validation library should we use

### DISCUSSION THREAD

**Discussion by mentalcaries** [Full-Stack-Collective/connectient#35] [Polls]
Right now, React-Hook-Form seems to be in the lead.
Are there any arguments for Formik?
Are there _any other_ better libraries?

**Comment by 5hraddha** [comment]
In  NextJS 13, they have something called [server actions](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions) and instead of using event handlers to submit the form and process the data, they are now suggesting to use server actions. 

Found a video - [Lee Robinson - Next.js Metamorphosis, React Summit 2023](https://www.youtube.com/watch?v=I9Ev044V4Lk) where Lee shares why Vercel made this shift to App Router. It's a good one. But, around [18.47 mins](https://youtu.be/I9Ev044V4Lk?t=1135), he starts to refactor code from 'old' NextJS to NextJS 13 and uses **Server Actions** instead of event handler to handle form data at server side.

Also, if you read the NextJS 13 Docs, using Server Actions will opt into running the React `experimental` channel. You can enable Server Actions in your Next.js project by enabling the experimental `serverActions` flag.  

```js
//next.config.js
module.exports = {
  experimental: {
    serverActions: true,
  },
}
```

The data passed to a Server Action can be validated or sanitized before invoking the action. For example, you can create a wrapper function that receives the action as its argument, and returns a function that invokes the action if it's valid. You could read about it - [here](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions#validation)

**Reply by 5hraddha** [reply]
Interesting discussion to follow on how to use server components with React Hook Forms - [here](https://github.com/react-hook-form/react-hook-form/issues/10391)
